manager/assets/modext/widgets/core/modx.grid.js

```diff
@@ -892,6 +930,82 @@ Ext.extend(MODx.grid.Grid,Ext.grid.EditorGridPanel,{
         MODx.util.url.clearParams();
         this.getBottomToolbar().changePage(1);
     }
+    
+    /**
+     * @property {Boolean} hasNestedFilters - Indicates whether the top toolbar filter(s) are nested
+     * within a secondary container; they will be nested when they have labels and those labels are
+     * positioned above the filter's input.
+     */
+    ,hasNestedFilters: false
+
+    /**
+     * @property {Function} getFilterComponent - Gets a filter component from the top toolbar by its itemId
+     *
+     * @param {String} filterId - The Ext itemId of the filter component to fetch
+     * @return {Ext.Component}
+     */
+    ,getFilterComponent: function(filterId) {
+        const topToolbar = this.getTopToolbar(),
+              cmp = this.hasNestedFilters && filterId !== 'filter-query'
+                ? topToolbar.find('itemId', `${filterId}-container`)[0].getComponent(filterId)
+                : topToolbar.getComponent(filterId)
+        ;
+        if (typeof cmp !== 'undefined') {
+            return cmp;
+        }
+        console.error(`getFilterComponent: The filter component with itemId '${filterId}' could not be retrieved.`);
+    }
+
+    /**
+     * @property {Function} refreshFilterOptions - Used to syncronize a filter's store options to those available in its target grid
+     *
+     * @param {Array} filterData - An array of objects containing info needed to refresh each filter
+     * @param {Boolean} clearDependentParams - If true, will clear values of dependentParams specified in the filterData
+     */
+    ,refreshFilterOptions: function(filterData = [], clearDependentParams = true) {
+        if (filterData.length > 0) {
+            filterData.forEach(data => {
+                const filter = this.getFilterComponent(data.filterId);
+                if (filter) {
+                    const store = filter.getStore();
+                    filter.setValue('');
+                    if (store) {
+                        if (data.hasOwnProperty('dependentParams')) {
```

Comment:
Code style wise, early returns help keep the code simple to understand and debug. This is 4 levels deep. No need to change that immediately as it's a nitpick, but for future consideration when implementing new features.